src/otx/algo/utils/xai_utils.py

```diff
@@ -47,31 +49,57 @@
 def process_saliency_maps_in_pred_entity(
     predict_result: list[OTXBatchPredEntitiesSupportXAI],
     explain_config: ExplainConfig,
+    label_info: LabelInfoTypes,
 ) -> list[OTXBatchPredEntitiesSupportXAI]:
     """Process saliency maps in PredEntity."""
 
-    def _process(predict_result_per_batch: OTXBatchPredEntitiesSupportXAI) -> OTXBatchPredEntitiesSupportXAI:
+    def _process(
+        predict_result_per_batch: OTXBatchPredEntitiesSupportXAI,
+        label_info: LabelInfoTypes,
+    ) -> OTXBatchPredEntitiesSupportXAI:
         saliency_map: list[np.ndarray] = [
             saliency_map.cpu().numpy() if isinstance(saliency_map, torch.Tensor) else saliency_map
             for saliency_map in predict_result_per_batch.saliency_map
         ]
         imgs_info = predict_result_per_batch.imgs_info
         ori_img_shapes = [img_info.ori_shape for img_info in imgs_info]
-        pred_labels = [pred.tolist() for pred in predict_result_per_batch.labels]
-
-        processed_saliency_maps = process_saliency_maps(saliency_map, explain_config, pred_labels, ori_img_shapes)
+        paddings = [img_info.padding for img_info in imgs_info]
+        image_shape = imgs_info[0].img_shape
+
+        pred_labels = []
+        for labels, scores in zip(predict_result_per_batch.labels, predict_result_per_batch.scores):
+            if isinstance(label_info, HLabelInfo):
+                pred_labels.append(_convert_labels_from_hcls_format(labels, scores, label_info))
+            elif labels.shape == scores.shape:
+                # Filter out predictions with scores less than 0.3
+                pred_labels.append(labels[scores > 0.3].tolist())
```

Comment:
It's a small thr just to filter out low-confidence predictions like 0.05-0.1, so it can be any number from `0.2` to `0.5`. I chose middle value and manually chose it works for test images.